recipes/gsl/all/conanfile.py

```diff
@@ -0,0 +1,158 @@
+
+required_conan_version = ">=1.33.0"
+
+class GslConan(ConanFile):
+    name = "gsl"
+    description = "GNU Scientific Library"
+    homepage = "https://www.gnu.org/software/gsl"
+    topics = ("numerical", "math", "random", "scientific")
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "GPL-3.0-or-later"
+
+    settings = "os", "compiler", "build_type", "arch"
+
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False]
+    }
+
+    default_options = {
+        "shared": False,
+        "fPIC": True
+    }
+
+    exports_sources = "patches/*"
+
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _settings_build(self):
+        return getattr(self, "settings_build", self.settings)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True,
+                  destination=self._source_subfolder)
+
+    def build_requirements(self):
+        self.build_requires("libtool/2.4.6")
+        if tools.os_info.is_windows and not tools.get_env("CONAN_BASH_PATH"):
+            self.build_requires("msys2/cci.latest")
+
+    @contextmanager
+    def _build_context(self):
+        if self.settings.compiler == "Visual Studio":
+            with tools.vcvars(self):
+                env = {
+                    "CC": "cl -nologo",
+                    "CXX": "cl -nologo",
+                    "LD": "link -nologo",
+                    "AR": "{} lib".format(tools.unix_path(self.deps_user_info["automake"].ar_lib)),
+                }
+                with tools.environment_append(env):
+                    yield
+        else:
+            yield
+
+    def _patch_source(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+        with tools.chdir(self._source_subfolder):
+            self.run("{} -fiv".format(tools.get_env("AUTORECONF")), win_bash=tools.os_info.is_windows, run_environment=True)
+        tools.replace_in_file(os.path.join(self._source_subfolder, "configure"),
+                              r"-install_name \$rpath/",
+                              "-install_name @rpath/")
+
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+        self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
+
+        self._autotools.libs = []
+        yes_no = lambda v: "yes" if v else "no"
+        args = [
+            "--enable-shared={}".format(yes_no(self.options.shared)),
+            "--enable-static={}".format(yes_no(not self.options.shared)),
```

Comment:
I haven't downloaded and checked the gsl sources, but sometimes autotools has a `--enable/disable-rpath` option.
```suggestion
            "--enable-static={}".format(yes_no(not self.options.shared)),
            "--disable-rpath",
```

Reply:
Ignore this suggestion. That option does not exist.